### ISSUE TITLE
Cleanup up Select Listeners

### ISSUE THREAD

**Issue by magnus-madsen** [flix/flix#1719]

**Comment by magnus-madsen** [comment]
@JonathanStarup Could you write a short description of what needs to be done?

**Comment by magnus-madsen** [comment]
@JonathanStarup Can we close this?

**Comment by JonathanStarup** [comment]
I should've written a short description, I guess I missed the ping. I think in general a lot of small things can be optimized on the channel impl so maybe we can list ideas once we have the new channel in place and make a issue with them all?

**Comment by magnus-madsen** [comment]
Is this fixed or?

**Comment by JonathanStarup** [comment]
I don't believe it is. I'm trying to think what the issue was. It seems I dropped the ball twice. 

I think the idea is that a select statement will (if in a blocking scenario) add a listener to each channel involved. If you do this repeatedly on a "slow" channel. Then a lot of dead listeners will be in queue for this element. Thus the select expression could remove its listeners after a successful retrieval on one of the other channels.

I'm not sure if this matters much for performance. It seems a bit like shifting work on `get` to equivalent work in `select` so I'm not sure if the complexity is worth it.

**Comment by magnus-madsen** [comment]
OK. Will leave open.

**Comment by magnus-madsen** [comment]
@JonathanStarup @paulbutcher Do we know if this is still relevant?

**Comment by JonathanStarup** [comment]
I'm leaning towards no, if someone works on channel performance at some point they will rediscover it

**Comment by magnus-madsen** [comment]
Ok

**Comment by mlutze** [comment]
Closing since Jonathan said it's not relevant.
